client/dashboard/domains/domain-connection-setup/utils/test/match-records.test.ts

```diff
@@ -0,0 +1,164 @@
+		const target = [ 'ns1.wordpress.com', 'ns2.wordpress.com', 'ns3.wordpress.com' ];
+
+		const result = matchCurrentToTargetValues( current, target );
+
+		expect( result ).toEqual( [
+			{ currentValue: 'ns1.wordpress.com', updateTo: 'ns1.wordpress.com' },
+			{ currentValue: 'ns2.wordpress.com', updateTo: 'ns2.wordpress.com' },
+			{ currentValue: 'ns3.wordpress.com', updateTo: 'ns3.wordpress.com' },
+		] );
+	} );
+
+	test( 'pairs unmatched current values with target values', () => {
+		const current = [ 'ns1.other.com', 'ns2.other.com', 'ns3.other.com' ];
+		const target = [ 'ns1.wordpress.com', 'ns2.wordpress.com', 'ns3.wordpress.com' ];
+
+		const result = matchCurrentToTargetValues( current, target );
+
+		expect( result ).toEqual( [
+			{ currentValue: 'ns1.other.com', updateTo: 'ns1.wordpress.com' },
+			{ currentValue: 'ns2.other.com', updateTo: 'ns2.wordpress.com' },
+			{ currentValue: 'ns3.other.com', updateTo: 'ns3.wordpress.com' },
+		] );
+	} );
+
+	test( 'uses "-" when current values are fewer than target values', () => {
+		const current = [ 'ns1.other.com', 'ns2.other.com' ];
+		const target = [ 'ns1.wordpress.com', 'ns2.wordpress.com', 'ns3.wordpress.com' ];
+
+		const result = matchCurrentToTargetValues( current, target );
+
+		expect( result ).toEqual( [
+			{ currentValue: 'ns1.other.com', updateTo: 'ns1.wordpress.com' },
+			{ currentValue: 'ns2.other.com', updateTo: 'ns2.wordpress.com' },
+			{ currentValue: '-', updateTo: 'ns3.wordpress.com' },
+		] );
+	} );
+
+	test( 'handles empty current values array', () => {
+		const current: string[] = [];
+		const target = [ 'ns1.wordpress.com', 'ns2.wordpress.com', 'ns3.wordpress.com' ];
+
+		const result = matchCurrentToTargetValues( current, target );
+
+		expect( result ).toEqual( [
+			{ currentValue: '-', updateTo: 'ns1.wordpress.com' },
+			{ currentValue: '-', updateTo: 'ns2.wordpress.com' },
+			{ currentValue: '-', updateTo: 'ns3.wordpress.com' },
+		] );
+	} );
+
+	test( 'prioritizes matched values over unmatched ones', () => {
+		const current = [ 'ns1.wordpress.com', 'ns2.other.com', 'ns3.other.com' ];
+		const target = [ 'ns1.wordpress.com', 'ns2.wordpress.com', 'ns3.wordpress.com' ];
+
+		const result = matchCurrentToTargetValues( current, target );
+
+		// ns1.wordpress.com should be matched first (same value)
+		expect( result[ 0 ] ).toEqual( {
+			currentValue: 'ns1.wordpress.com',
+			updateTo: 'ns1.wordpress.com',
+		} );
+
+		// Remaining unmatched values should be paired in order
+		expect( result[ 1 ] ).toEqual( {
+			currentValue: 'ns2.other.com',
+			updateTo: 'ns2.wordpress.com',
+		} );
+		expect( result[ 2 ] ).toEqual( {
+			currentValue: 'ns3.other.com',
+			updateTo: 'ns3.wordpress.com',
+		} );
+	} );
+
+	test( 'handles duplicate target values correctly', () => {
+		const current = [ 'ns1.wordpress.com', 'ns1.wordpress.com' ];
+		const target = [ 'ns1.wordpress.com', 'ns1.wordpress.com' ];
```

Comment:
Would this ever happen? 🤔 It doesn't hurt having a test, but I wonder if this is possible 🙂